go/nbs/table_reader.go

```diff
@@ -38,21 +40,28 @@ func newTableReader(buff []byte, r io.ReaderAt, readAmpThresh uint64) tableReade
 	pos -= uint64Size
 
 	pos -= uint32Size
-	tr.chunkCount = binary.BigEndian.Uint32(buff[pos:])
+	chunkCount := binary.BigEndian.Uint32(buff[pos:])
 
 	// index
-	suffixesSize := uint64(tr.chunkCount) * addrSuffixSize
+	suffixesSize := uint64(chunkCount) * addrSuffixSize
 	pos -= suffixesSize
-	tr.suffixes = buff[pos : pos+suffixesSize]
+	suffixes := make([]byte, suffixesSize)
+	copy(suffixes, buff[pos:])
```

Comment:
Per my comment in slack, "I do also wonder if the fact that different approaches are ideal for different persistence layers (e.g. mmap the index for files, copying for S3) means that we should have different code pathways for creating the tableReader"

There's no need to copy this data over when backed by files, though there may be a need when the data is already in memory.